java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -2219,6 +2221,79 @@ void testGroupBySum() {
     }
   }
 
+  @Test
+  void testGroupByFirstExcludeNulls() {
+    try (Table t1 = new Table.TestBuilder()
+            .column(  1,   1,    1,  1,  2,    2,  2,    2)
+            .column(null, 13, null, 12, 14, null, 15, null)
+            .build()) {
+
+      try(Table t2 = t1.groupBy(0)
+              .aggregate(
+                      first(1, false));
+          HostColumnVector firstExcludeNulls = t2.getColumn(1).copyToHost()) {
+
+        assertEquals(2, firstExcludeNulls.getRowCount());
+        assertEquals(13, firstExcludeNulls.getInt(0));
+        assertEquals(14, firstExcludeNulls.getInt(1));
+      }
+    }
+  }
+
+  @Test
+  void testGroupByLastExcludeNulls() {
```

Comment:
Similar here we can make the test a lot simpler.

Reply:
Done